lib/ansible/modules/windows/win_nssm.ps1

```diff
@@ -403,7 +403,7 @@ Function Nssm-Update-Credentials
             }
 
             If ($nssm_result.stdout.split("`n`r")[0] -ne $fullUser) {
-                $cmd = "set ""$name"" ObjectName $fullUser '$password'"
+                $cmd = "set ""$name"" ObjectName $fullUser ""$password"""
```

Comment:
Sorry for taking so long to get to review this PR. We have a better way to easily escape a list of arguments so that they are formatted properly when running the command. You can use the cmdlet `$cmd = Argv-ToString -arguments @("set", $name, "ObjectName", $fullUser, $password)`.

The `Argv-ToString` cmdlet is a module util that's provided by https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/powershell/Ansible.ModuleUtils.ArgvParser.psm1. It takes in a list of arguments and produces a string that has escaped the characters that need to be escaped.

The module also already imports this util with https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_nssm.ps1#L9 so you don't have to do that as well :).

Reply:
Updated the line to use the Argv-ToString function. Did some minimal testing with the below playbook which seemed to work as intended.

```
---
- hosts:
    - all
  vars:
    testuser_password: Password1234##
  tasks:
    - name: Install NSSM
      win_chocolatey:
        name: "nssm"

    - name: Create test user
      win_user:
        name: testuser
        password: "{{ testuser_password }}"

    - name: Create service script
      win_copy:
        dest: "C:\\Users\\service.bat"
        content: |
          pause
      notify: Restart TestService

    - name: Install service via NSSM
      win_nssm:
        name: TestService
        application: "C:\\Users\\service.bat"
        start_mode: auto
        user: testuser
        password: "{{ testuser_password }}"
      notify: Restart TestService

  handlers:
    - name: Restart TestService
      win_service:
        name: TestService
        state: restarted
```